### PR TITLE
feat: add new zone allowance options & fix potential exploit

### PR DESCRIPTION
**Pull request by TonybynMp4**
## Description

* Added the possibility to blacklist / whitelist classes & blacklist models.

* Fixed an exploit where if you were allowed to enter the customs and the conditions changed (before this PR it would be if you 
  got fired while in the zone) you could still access the customs.
  With the new class / model checks that would make it exploitable so i extracted the check to also check on enter/exit vehicle.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/zones.lua [hunk 1]
```diff
@@ -2,6 +2,38 @@ local zoneId
...
+
+    if (zone.deniedClasses and zone.deniedClasses[vehicleClass]) or (zone.allowedClasses and not zone.allowedClasses[vehicleClass]) then
+        allowAccess = false
+        return
+    end
+
+    if zone.modelBlacklist and zone.modelBlacklist[GetEntityModel(cache.vehicle)] then
+        allowAccess = false
+        return
+    end
```
Comment (BerkieBb) [inline client/zones.lua]: These can all be one if statement

File: client/zones.lua [hunk 2]
```diff
@@ -2,6 +2,38 @@ local zoneId
...
+
+    if zone.job and QBX?.PlayerData then
+        hasJob = false
+        local playerJob = QBX.PlayerData.job.name
+        for _, job in ipairs(zone.job) do
```
Comment (BerkieBb) [inline client/zones.lua]: Can just be a for i = ...

File: client/zones.lua [hunk 3]
```diff
@@ -54,10 +75,18 @@ CreateThread(function()
...
+                    if not lib.isTextUIOpen() then
+                        lib.showTextUI(locale('textUI.tune'), {
+                            icon = 'fa-solid fa-car',
+                            position = 'left-center',
+                        })
+                    end
+                    if IsControlJustPressed(0, 38) then
+                        SetEntityVelocity(cache.vehicle, 0.0, 0.0, 0.0)
+                        lib.hideTextUI()
+                        require('client.menus.main')()
```
Comment (BerkieBb) [inline client/zones.lua]: Why require each time it is pressed?

Comment (TonybynMp4) [reply client/zones.lua]: idk ask jorn 😆

**Review by BerkieBb — APPROVED** [review]

**Review by FjamZoo — APPROVED** [review]

**Pull request merged**